### PR TITLE
Move revalidate mechanism to request promise creator

### PR DESCRIPTION
**Pull request by oliverlin**
Since the `revalidate` parameter is added, the action creators of `CHAIN_API` are executed with the missing response which caused the unexpected behavior on the consumers. 
Fix it by moving the `revalidate` from `index.js` to `createRequestPromise` to prevent executing it twice.

Related slack bug report https://codementor.slack.com/archives/C7YGG44Q1/p1613138346001100

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -39,22 +38,6 @@ export default ({
 
     return new Promise((resolve, reject) => {
       const promiseCreators = action[CHAIN_API].map((createCallApiAction) => {
-        const apiAction = createCallApiAction()[CALL_API]
```
Comment (ocowchun) [inline src/index.js:42]: The key change is don't call `createCallApiAction` here, the actions should only be called on https://github.com/CodementorIO/redux-api-middleman/blob/598d554f5691e96f4bebd3d7b3b76bafbec1126e/src/index.js#L54

**Review by ocowchun — APPROVED** [review]

**Comment by oliverlin** [comment]
@ocowchun Thanks for reviewing! 
I made one more change -> disable revalidate for `CHAIN_API` actions. Thanks for pointing this out. I think for now not to support revalidate on CHAIN_API actions should satisfy our needs.
Please review this [newly added commit](https://github.com/CodementorIO/redux-api-middleman/pull/39/commits/b9db070c78ba29f53f5cd30e73fed880a399e3f2) 🙏

File: src/index.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export default ({
   return ({ dispatch, getState }) => next => action => {
     if (action[CALL_API]) {
       return dispatch({
+        isNativeCallApi: true,
```
Comment (ocowchun) [inline src/index.js]: `revalidateDisabled: !action.isNativeCallApi` is a bit counterintuitive. How about changing the key to `revalidateDisabled` directly?

Comment (oliverlin) [reply src/index.js]: Was thinking a way to separate if the action is the original `CALL_API` or `CHAIN_API` but I think it makes sense to make it consistent in this case. Thanks!

**Pull request merged**